src/wrover_status_light/ServiceIntro.md

```diff
@@ -0,0 +1,66 @@
+* Server node: the server node receives and handles the service request calls from client nodes. It retrieves data from the request message, performs the necessary actions, and generates a response message which is sent back to the client. 
+* Request and response messages: these messages are structured data containers that are defined by the service definition. The client and server use these messages to exchange information.
+
+## Why do we use Services when there's already Topics?
+
+While both services and topics work to allow communication between nodes, they differ in terms of their data exchange patterns and the types of interactions they support. 
+
+Unlike topics, which operate asynchronously, service calls provide synchronous communication. The client node waits for the server node to process the request and receive a response before continuing its execution. This synchronous behavior is beneficial in cases where a response is required before proceeding with the next steps of a program.  It is not acceptable for a server to not exist or to never return a response; these will generate exceptions for the service client to handle.
+
+In addition, topics are designed for data streaming and broadcasting continuous streams of information, such as sensor data or robot state updates. They are ideal when nodes require a constant flow of information and do not require immediate responses. On the other hand, services are typically used for accessing functionality that is provided by another node. They are well-suited for situations where nodes need to invoke specific actions or request information from another node, and a response is expected.
+
+## Defining a Service
+Services are defined by srv files, which are identical to msg files in terms of formatting, and should be located in the `/srv` folder in a package. In this example, the service takes in a `string` and an `int` as input and returns a `float` as output. You can find an example service file at `srv/DoubleService.srv`.
+```
+#request fields
+string input1
+int32 input2
+---
+#response fields
+float32 output
+```
+After defining a message and building this project, ROS would convert a srv file into three classes: service definition, request messages, and response messages. Suppose we have `ExampleService.srv` in an package called `ExamplePackage`, then the names of these packages would be:
+|                   | C++                                     | Python                                    |
+|---                 | ---                                     | ---   |
+| Service definition |`ExamplePackage::ExampleService`          |`ExamplePackage.srv.ExampleService`        |
+| Request message    |`ExamplePackage::ExampleService::Request` |`ExamplePackage.srv.ExampleServiceRequest` |
+| Response message   |`ExamplePackage::ExampleService::Response`|`ExamplePackage.srv.ExampleServiceResponse`|
+
+## Writing Server and Client nodes
```

Comment:
A general description of what the demo code does would be nice here. It would also be good to explain the format of the .srv file either in here, or in comments.

Reply:
Addressed in commit [c71a540](https://github.com/WisconsinRobotics/WRoverPlayground/pull/9/commits/c71a54021d2c6df684d511cbd4e7e1915db39a43).